execution/aggregate/khashaggregate.go

```diff
@@ -91,19 +91,25 @@ func (a *kAggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 	if err != nil {
 		return nil, err
 	}
-	for i := range a.params {
-		a.params[i] = math.NaN()
-		if i < len(args) {
-			a.params[i] = args[i].Samples[0]
-			a.paramOp.GetPool().PutStepVector(args[i])
-		}
-	}
-	a.paramOp.GetPool().PutVectors(args)
+	defer a.paramOp.GetPool().PutVectors(args)
```

Comment:
Using `defer` to release vectors has lead to some subtle bugs in the past. Could we revert to the previous implementation where we do it explicitly after iterating through parameter vectors?

Reply:
I thought it would be better like this since we could return early and in that case we would not return to the pool ( or explicitely `Put` on every return statement ). Out of curiosity what was the bug with defer?

Reply:
reverted to putting the vectors back after the loops